ground/src/main/java/com/google/android/ground/ui/surveyselector/UiState.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.surveyselector
+
+import com.google.android.ground.model.SurveyListItem
+
+sealed class UiState {
+
+  data object FetchingSurveys : UiState()
+
+  data class SurveyListAvailable(val surveys: List<SurveyListItem>) : UiState()
+
+  data object ActivatingSurvey : UiState()
+
+  data object SurveyActivated : UiState()
+
+  data class Error(val errorResId: Int) : UiState()
```

Comment:
how about naming these objet and class in such a way where it can represent a state as it is part of UiState? 

example - `FetchingSurveys` => `FetchSurveys`, sounds more actionable, WDYT?

Reply:
I think gerund is ok here, if the state is emphasizing something that's ongoing.

Since the state is specified to the survey list, I think we can omit "Survey" simply and say `LoadingList`.

Also, `SurveyActivated` sounds more like an event than a UI state. What does it represent? 

On that note, please add ktdoc to describe what each of these states represents.